packages/rest/tests/basicMessage.test.ts

```diff
@@ -9,46 +10,72 @@ import { getTestAgent, objectToJson } from './utils/helpers'
 
 describe('BasicMessageController', () => {
   let app: Express
-  let agent: Agent
-  let connection: ConnectionRecord
+  let aliceAgent: Agent
+  let bobAgent: Agent
+  let bobConnectionToAlice: ConnectionRecord
+  let aliceConnectionToBob: ConnectionRecord
 
   beforeAll(async () => {
-    agent = await getTestAgent('rest agent basic message test', 3011)
-    app = await setupServer(agent, { port: 3000 })
-    const { invitation } = await agent.connections.createConnection()
-    connection = await agent.connections.receiveInvitation(invitation)
+    aliceAgent = await getTestAgent('REST Agent Test Alice', 3012)
+    bobAgent = await getTestAgent('REST Agent Test Bob', 3013)
+    app = await setupServer(bobAgent, { port: 3000 })
   })
 
-  describe('get basic messages', () => {
-    test('should return list of basic messages filtered by connectionId', async () => {
-      const spy = jest.spyOn(agent.basicMessages, 'findAllByQuery')
-      const getResult = (): Promise<BasicMessageRecord[]> => spy.mock.results[0].value
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
 
-      const response = await request(app).get(`/basic-messages/${connection.id}`)
-      const result = await getResult()
+  describe('Alice and Bob connect', () => {
+    test('make a connection between agents', async () => {
+      const { outOfBandInvitation, ...aliceOOBRecord } = await aliceAgent.oob.createInvitation()
 
-      expect(response.statusCode).toBe(200)
-      expect(response.body).toEqual(result.map(objectToJson))
+      const { outOfBandRecord: bobOOBRecord } = await bobAgent.oob.receiveInvitation(outOfBandInvitation)
+      const [bobConnectionAtBobAlice] = await bobAgent.connections.findAllByOutOfBandId(bobOOBRecord.id)
+      bobConnectionToAlice = await bobAgent.connections.returnWhenIsConnected(bobConnectionAtBobAlice!.id)
+
+      const [aliceConnectionAtAliceBob] = await aliceAgent.connections.findAllByOutOfBandId(aliceOOBRecord.id)
+      aliceConnectionToBob = await aliceAgent.connections.returnWhenIsConnected(aliceConnectionAtAliceBob!.id)
+
+      expect(aliceConnectionToBob?.theirDid).toEqual(bobConnectionToAlice?.did)
+      expect(bobConnectionToAlice?.theirDid).toEqual(aliceConnectionToBob?.did)
     })
   })
```

Comment:
Actually, good point. On further thought should we create some sort of a reusable setup function, since it seems a majority of tests have Alice and Bob interactions?

Right on, re. the testing boundary 👍.

Reply:
Good idea. 👍